AdvLoggerPkg/Application/DecodeUefiLog/DecodeUefiLog.py

```diff
@@ -293,27 +293,27 @@ class AdvLogParser ():
     # #define DEBUG_ERROR  0x80000000          // Error
 
     debug_levels_dict = {

```

Comment:
Would it be easier to change how these strings are used instead of changing the string themselves?

Reply:
I don't think they should be defined with a space. That is a formatting decision that should be made at formatting time.

Reply:
@liqiqiii, can you describe the issue you encountered trying to change this here?

Reply:
I was seeing a red null when trying to open the output log file in vscode:

I was trying to do this:
  // Print the debug levels
  for (Index = 0; Index < ARRAY_SIZE (DebugLevels); Index++) {
    if ( (DebugLevel & DebugLevels[Index].Value) == DebugLevels[Index].Value) {
      DebugLevelStringLen = AsciiSPrint (MessageBuffer, MessageBufferSize, DebugLevels[Index].Name);
      // Add an extra space to the end of the string for readability purposes
      AsciiSPrint(MessageBuffer + DebugLevelStringLen, 1, " ");
      // AsciiSPrint(MessageBuffer + DebugLevelStringLen + 1, 1, "\0");  //with and without this line the same red null
      DebugLevelStringLen++;
      return (UINT16)DebugLevelStringLen;
    }

Reply:
Can you try this?

```c
  if (DebugLevelStringLen < MessageBufferSize - 1) {
    MessageBuffer[DebugLevelStringLen] = ' ';
    MessageBuffer[DebugLevelStringLen + 1] = '\0';
  }
```

Reply:
It works! It works without using the AsciiSPrint.
      // Verify string length and add an extra space
      if (DebugLevelStringLen < MessageBufferSize - 1) {
        MessageBuffer[DebugLevelStringLen]     = ' ';
        MessageBuffer[DebugLevelStringLen + 1] = '\0';
        DebugLevelStringLen++;
      }
When I tried the same thing with the AsciiSPrint it will still have the red null and that's where I stuck, no matter whether I add only a ' ' or also ' ' + '\0' to the end it will always have the red null, so I think the function did sth that will cause it.
Thanks so much Michael.

Reply:
It might be relevant to the input of AsciiSPrint, it needs to send in a string instead of a char, so " " might be causing the issue. Although theorectically it should be the same when I only copy the first char in the string, that's why I didn't think it is the problem.